Tests/Integration/Components/ProductToShopTest.php

```diff
@@ -83,6 +85,209 @@ public function prepare()
         );
     }
 
+    public function test_insert_variants()
```

Comment:
if you extract some test from legacy tests, remove it after that.

Reply:
because of the merge fuckUp in productToShop i'm currently happy that we have the ProductToShop double checked, i don't remove it for now
but we have to generally take care of the legacytests later